### DISCUSSION TITLE
[iOS] Integration MapTiler - how to change the User Agent ?

### DISCUSSION THREAD

**Discussion by nashfive** [maplibre/maplibre-native#2865] [Q&A]
Hi!
I am using MapLibre with MapTiler and they offer a way to secure their API key by doing some filtering the user-agent HTTP header.
I couldn't find any way to update the User-Agent for the requests that the MapLibre SDK is sending :/ 
I tried to add this in my application delegate that implements the `MLNNetworkConfigurationDelegate`: 

```
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
    MLNNetworkConfiguration.sharedManager.delegate = self
    return true
}

func session(for configuration: MLNNetworkConfiguration) -> URLSession {
    let conf = URLSessionConfiguration.default
    conf.httpAdditionalHeaders = ["User-Agent": "my-custom-string"]
    return URLSession(configuration: conf)
}
```

This is called, but the User-Agent is still the default one:
`User-Agent: TestMapLibre/1.0 MapLibre/6.6.0 MapLibreNative/0.0.0 (NOT SET) iOS Simulator/18.0.0 (arm64)``

How can we update this HTTP field?

**Comment by louwers** [comment]
I think this is possible, let me have a look.

**Comment by louwers** [comment]
@nashfive Can you try this https://github.com/maplibre/maplibre-native/pull/2293 ?

**Comment by nashfive** [comment]
@louwers Hi Bart, thanks for looking into this, but the ticket you linked doesn't help. I had seen it too when searching the issues, but it appears that MLNNetworkManager is not public, so I can't access it from my code... So by looking into the source code, I can see that it is used (see `platform/darwin/src/http_file_source.mm`, line 90) but the SDK currently doesn't allow us to access it directly...
Hence my question :)

**Reply by louwers** [reply]
You can use`MLNNetworkConfiguration` to configure the `NSURLSessionConfiguration` that is used globally.

**Reply by nashfive** [reply]
That's exactly what I've done (see the code in my initial post), but it doesn't work (the User-Agent is stuck to the default one provided by the SDK) and I can't figure out the way to do it from the documentation nor from the source. 😢

**Reply by louwers** [reply]
Oh, right. Looks right to me.

@hactar do you see an issue with the code snippet?
